lib/Conversion/TorchToLinalg/Uncategorized.cpp

```diff
@@ -1971,6 +1974,81 @@ class ConvertTensorStaticInfoCastOp
 };
 } // namespace
 
+namespace {
+class ConvertLogitOp : public OpConversionPattern<AtenLogitOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(AtenLogitOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    Location loc = op->getLoc();
+    Value input = adaptor.getSelf();
+    Value eps = adaptor.getEps();
+
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+      return failure();
+
+    bool handleEps = false;
+    if (succeeded(checkNotNone(rewriter, op, eps)))
+      handleEps = true;
+
+    auto inputType = input.getType().cast<RankedTensorType>();
+    auto inputElementType = inputType.getElementType();
+
+    auto inputRank = inputType.getRank();
+
+    SmallVector<AffineMap> indexingMaps = {
+        rewriter.getMultiDimIdentityMap(inputRank), // input
+        rewriter.getMultiDimIdentityMap(inputRank), // output
+    };
+    SmallVector<utils::IteratorType> iteratorTypes(
+        inputRank, utils::IteratorType::parallel);
+    Value logit =
+        rewriter
+            .create<linalg::GenericOp>(
+                loc, input.getType(),
+                /*ins=*/input,
+                /*outs=*/input,
+                /*indexingMaps=*/indexingMaps,
+                /*iteratorTypes=*/iteratorTypes,
+                [&](OpBuilder &b, Location loc, ValueRange args) {
+                  Value input = args[0];
+
+                  TypedAttr oneAttr = b.getFloatAttr(inputElementType, 1.0);
+                  Value oneValue = b.create<arith::ConstantOp>(loc, oneAttr);
+
+                  Value zI;
+                  if (!handleEps) {
+                    zI = input;
+                  } else {
+                    Value truncEps =
+                        b.create<arith::TruncFOp>(loc, inputElementType, eps);
+                    Value oneMinusEps =
+                        b.create<arith::SubFOp>(loc, oneValue, truncEps);
+
+                    Value min =
+                        b.create<arith::MinimumFOp>(loc, input, oneMinusEps);
+                    Value clampedInput =
+                        b.create<arith::MaximumFOp>(loc, min, truncEps);
+
```

Comment:
Are you sure that these values will always be Floating-point values? Otherwise ops like `...FOp` won't work.

Reply:
The logit (https://pytorch.org/docs/stable/special.html#torch.special.logit) function is defined as:
```math
\begin{align}
    y_i &= \ln({z_i \over 1 - z_i}) \\
    z_i &= \begin{cases}
        x_i & \text{if eps is None} \\
        eps & if x_i < eps \\
        x_i & if eps \leq x_i \leq 1 - eps \\
        1 - eps & if x_i > 1 - eps
    \end{cases}
\end{align}
```
So this means that the input should be a floating point value in the range of `[0, 1)` or clamped to `[eps, 1-eps]` if `eps` is defined.

Reply:
I tried the PyTorch op with integer inputs, and it didn't throw a runtime error. You should add a check at the top restricting the inputs to be of Floating-point type.

Reply:
Got it, I thought the type checks were implicit, thanks!

Also, why must the `ElementwiseLogitModule_basic` be put in xfail_sets?

Reply:
There are several configs for which the e2e tests are run. This test is failing for the LTC backend and for that reason you need to add it to the `LTC_XFAIL_SET`.